tests/installation/welcome.pm

```diff
@@ -18,38 +18,50 @@ use testapi;
 use utils 'ensure_fullscreen';
 
 sub run() {
-    while (1) {
-        my @welcome_tags = ('inst-welcome-confirm-self-update-server', 'scc-invalid-url');
-        if (get_var('BETA')) {
-            push @welcome_tags, 'inst-betawarning';
-        }
-        else {
-            push @welcome_tags, 'inst-welcome';
-        }
-        ensure_fullscreen;
-        assert_screen \@welcome_tags, 500;
+    my $iterations;
+
+    my @welcome_tags = ('inst-welcome-confirm-self-update-server', 'scc-invalid-url', 'inst-welcome');
+    if (get_var('BETA')) {
+        push @welcome_tags, 'inst-betawarning';
```

Comment:
Tried to clean up the code a bit, as we process required things and specific scenarios in one place. Now moved assertions out of the loop, as we would need to do extra assertions to fail with relevant error in case of too many iterations. The only problem now we have extra assertion if exited loop normally, but code is more readable imho.